test/node-api.js

```diff
@@ -6,11 +6,188 @@ chai.use(codeEquality);
 
 const { expect } = chai;
 
-const p = new Preprocessor();
+describe("parse", function() {
+  it("basic example", function() {
+    let p = new Preprocessor();
+    let output = p.parse("<template>Hello!</template>");
 
-describe("something", function () {
-  it("works for a basic example", function () {
-    let output = p.process("<template>Hi</template>");
+    expect(output).to.eql([
+      {
+        type: "template-tag",
+        tagName: "template",
+        contents: "Hello!",
+        range: {
+          start: 0,
+          end: 27
+        },
+        contentRange: {
+          start: 10,
+          end: 16
+        },
+        startRange: {
+          end: 10,
+          start: 0
+        },
+        endRange: {
+          start: 16,
+          end: 27
+        }
+      },
```

Comment:
The API is slightly different from the existing one, in that we normalized all the ranges to `{ start: number, end: number }`, whereas the existing API uses `[start: number, end: number]` for some seemingly randomly